lib/trino-hive-formats/src/main/java/io/trino/hive/formats/encodings/text/BooleanEncoding.java

```diff
@@ -102,24 +104,36 @@ else if (isFalse(slice, offset, length)) {
     }
 
     @SuppressWarnings("PointlessArithmeticExpression")
-    private static boolean isFalse(Slice slice, int start, int length)
+    private boolean isFalse(Slice slice, int start, int length)
     {
-        return (length == 5) &&
-                (toUpperCase(slice.getByte(start + 0)) == 'F') &&
-                (toUpperCase(slice.getByte(start + 1)) == 'A') &&
-                (toUpperCase(slice.getByte(start + 2)) == 'L') &&
-                (toUpperCase(slice.getByte(start + 3)) == 'S') &&
-                (toUpperCase(slice.getByte(start + 4)) == 'E');
+        if (length == 5) {
+            return (toUpperCase(slice.getByte(start + 0)) == 'F') &&
+                    (toUpperCase(slice.getByte(start + 1)) == 'A') &&
+                    (toUpperCase(slice.getByte(start + 2)) == 'L') &&
+                    (toUpperCase(slice.getByte(start + 3)) == 'S') &&
+                    (toUpperCase(slice.getByte(start + 4)) == 'E');
+        }
+        else if (length == 1 && isExtendedBooleanLiteral) {
+            return toUpperCase(slice.getByte(start + 0)) == 'F' ||
```

Comment:
Can we store the result of  `getByte` to a variable and reuse them ?

Reply:
done. thank you for the review.